.golangci.yml

```diff
@@ -35,37 +51,233 @@ linters:
   - unconvert
   - unparam
   - unused
+  - usestdlibvars
   - whitespace
 
 linters-settings:
   gci:
-    local-prefixes: sigs.k8s.io/cluster-api-provider-vsphere
-run:
-  skip-files:
-  - ".*zz_generated.*\\.go"
-  - "contrib/.*"
-  - "apis/v1alpha3/.*"
-  - "apis/v1alpha4/.*"
-  timeout: 5m
-issue:
-  max-same-issues: 0
-  max-per-linter: 0
+    local-prefixes: "sigs.k8s.io/cluster-api-provider-vsphere"
+  godot:
+    #   declarations - for top level declaration comments (default);
+    #   toplevel     - for top level comments;
+    #   all          - for all comments.
+    scope: toplevel
+    exclude:
+      - '^ \+.*'
+      - '^ ANCHOR.*'
+  # gocritic:
+  #   enabled-tags:
+  #     - diagnostic
+  #     - experimental
+  #     - performance
+  #   disabled-checks:
+  #     - appendAssign
+  #     - dupImport # https://github.com/go-critic/go-critic/issues/845
+  #     - evalOrder
+  #     - ifElseChain
+  #     - octalLiteral
+  #     - regexpSimplify
+  #     - sloppyReassign
+  #     - truncateCmp
+  #     - typeDefFirst
+  #     - unnamedResult
+  #     - unnecessaryDefer
+  #     - whyNoLint
+  #     - wrapperFunc
+  #     - rangeValCopy
+  #     - hugeParam
```

Comment:
Should we also have this?

```
  gosec:
    excludes:
    - G307 # Deferring unsafe method "Close" on type "\*os.File"
    - G108 # Profiling endpoint is automatically exposed on /debug/pprof
```

Can you check if we still need the second one for core CAPI? I think this is not necessary since we moved to using the CR builtin profile server.

I would have expected that we have some cases of G307 in CAPV, but if we don't let's not add the exclude of course

Reply:
Maybe we also don't need the first one anymore in core CAPI. I think we maybe should prefer the excludes slice with a per-linter exclude over this (if easily possible)

Reply:
see https://github.com/kubernetes-sigs/cluster-api/pull/9156